src/consumption.cpp

```diff
@@ -1095,6 +976,134 @@ void Character::modify_morale( item &food, const int nutr )
         add_morale( MORALE_FOOD_GOOD, fun.first, fun.second, morale_time, morale_time / 2, false,
                     food.type );
     }
+
+    bool spoiled = food.rotten();
+
+    // Morale bonus for eating unspoiled food at a nearby table
+    if( !spoiled && !food.has_flag( flag_ALLERGEN_JUNK ) ) {
+        bool has_table_nearby = false;
+        bool has_chair_nearby = false;
+        for( const tripoint &pt : g->m.points_in_radius( pos(), 1 ) ) {
+            if( g->m.has_flag_furn( flag_FLAT_SURF, pt ) || g->m.has_flag( flag_FLAT_SURF, pt ) ||
+                ( g->m.veh_at( pt ) && ( g->m.veh_at( pt )->vehicle().has_part( "KITCHEN" ) ||
+                                         g->m.veh_at( pt )->vehicle().has_part( "FLAT_SURF" ) ) ) ) {
```

Comment:
`veh_at` isn't cost-free under the hood, not mentioning repetition. Just extract a temporary variable and reuse it five times.

Reply:
Fixed in https://github.com/CleverRaven/Cataclysm-DDA/pull/37713/commits/0f5b15d7820aa0ecb1df51b494fea45153ddaad9